packages/frontend/editor-ui/src/features/dataStore/components/dataGrid/DataStoreTable.vue

```diff
@@ -288,21 +335,66 @@ const initColumnDefinitions = () => {
 	];
 };
 
-const onCellValueChanged = async (params: CellValueChangedEvent) => {
-	const { data, api } = params;
+const onCellValueChanged = async (params: CellValueChangedEvent<DataStoreRow>) => {
+	const { data, api, oldValue, value, colDef } = params;
+
+	if (value === oldValue) {
+		return;
+	}
+
+	if (typeof data.id !== 'number') {
+		throw new Error('Expected row id to be a number');
+	}
+	const fieldName = String(colDef.field);
+	const id = data.id;
 
 	try {
 		emit('toggleSave', true);
-		await dataStoreStore.upsertRow(props.dataStore.id, props.dataStore.projectId, data);
+		await dataStoreStore.updateRow(props.dataStore.id, props.dataStore.projectId, id, {
+			[fieldName]: value,
+		});
 	} catch (error) {
 		// Revert cell to original value if the update fails
-		api.undoCellEditing();
+		const validOldValue = isDataStoreValue(oldValue) ? oldValue : null;
```

Comment:
Is it possible for the oldValue to be invalid?

Reply:
Probably not, but keeping the typescript happy with this